### PR TITLE
Adding file format network/rtpdump

### PR DESCRIPTION
**Pull request by julienblitte**
Popular file format for RTP tools to replay streams (telephony, audio, video) from network capture.
http://www.cs.cmu.edu/~./libra-demo/rtptools-1.17/rtptools.html

### CODE REVIEW

File: network/rtpdump.ksy [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - id: space
+        contents: ' '
+      - id: ip
+        type: str
+        encoding: ascii
+        terminator: 0x2f // '\'
+      - id: port
+        type: str
+        encoding: ascii
+        terminator: 0x0a // '\n'
```
Comment (generalmimon) [inline network/rtpdump.ksy:36]:
```suggestion
        terminator: 0x0a # '\n'
```

File: network/rtpdump.ksy [hunk 2]
```diff
@@ -0,0 +1,73 @@
+meta:
+  id: rtmpdump
+  title: Rtpdump (rtptools)
+  file-extension: rtmpdump
+  license: Unlicense
+  ks-version: 0.1
```
Comment (generalmimon) [inline network/rtpdump.ksy:6]:
```suggestion
```

I'd rather leave this out, it doesn't really have any benefit, given that it doesn't limit the minimum KS version in any way. And declaring version 0.1 is just weird.

File: network/rtpdump.ksy [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        type: u2
+        doc: |
+          2 bytes padding.
+  packet_t:
+    seq:
+      - id: length
+        type: u2
+        doc: |
+          packet length (including this header).
+      - id: plen
```
Comment (generalmimon) [inline network/rtpdump.ksy:63]:
```suggestion
      - id: len_body
```

It's recommended to stick to the "standard" naming specified in the [KSY style guide] (http://doc.kaitai.io/ksy_style_guide.html#attr-id). For example, if some integer denotes the byte size of an attribute  `item` (i.e. this integer will be used in the `size` key of the `item`), it's recommended to name it `len_item`.

File: network/rtpdump.ksy [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        type: u2
+        doc: |
+          payload length.
+      - id: packet_usec
+        type: u4
+        doc: |
+          timestamp of packet since the start.
+      - id: body
+        size: plen
```
Comment (generalmimon) [inline network/rtpdump.ksy:72]:
```suggestion
        size: len_body
```

File: network/rtpdump.ksy [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  rtpdump is a format used by rtptools to record and replay
+  rtp data from network capture.
+doc-ref: https://chromium.googlesource.com/external/webrtc/stable/talk/+/master/media/base/rtpdump.h
+seq:
+  - id: file_header
+    type: header_t
+  - id: packets
+    type: packet_t
+    repeat: eos
+-includes:
```
Comment (generalmimon) [inline network/rtpdump.ksy:20]: Please don't use the dash-prefixed properties in the official KS format gallery, especially when they just mirror what's already present in `meta/imports`.

Comment (julienblitte) [reply network/rtpdump.ksy]: I was not sure if it was relevant to have both "includes" and "imports".
Remove the dash-prefixed as instructed on another commit.

File: network/rtpdump.ksy [hunk 6]
```diff
@@ -0,0 +1,70 @@
+meta:
+  id: rtmpdump
+  title: Rtpdump (rtptools)
+  file-extension: rtmpdump
```
[the remark this answers is not in the document]
Comment (julienblitte) [reply network/rtpdump.ksy:4]: Thanks.

Before answering this, I see typos I made:
- The id should be `rtpdump` not ~`rtmpdump`~;
- Same, the file-extension I wanted originally was `rtpdump`.

Now, regarding the extension, most of modern tools in the industries (commercial tools) use the `.rtpdump` extension as well as the modern open source tool Wireshark.

See an example of export on Wireshark, the most used tool today to generate such files:
image

Using only `rtp` as extension would look a bit conservative as it sticks to the command line rtptools recommendation but differs from people's custom today.

I commit a proposed fix to add the two extensions (original `rtp` and `rtpdump` widely in use today). I hope it is fine for your side.

**Comment by GreyCat** [comment]
Thanks for great work, @julienblitte and @generalmimon for reviewing!

**Pull request merged**